Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponse/ProofpointThreatResponse.py

```diff
@@ -494,38 +551,52 @@ def fetch_incidents_command():
     """
     integration_params = demisto.params()
     last_fetch = demisto.getLastRun().get('last_fetch', {})
+    last_fetched_id = demisto.getLastRun().get('last_fetched_incident_id', {})
+
+    fetch_delta = integration_params.get('fetch_delta', '6')
+    fetch_limit = integration_params.get('fetch_limit', '50')
+
     incidents_states = integration_params.get('states')
     for state in incidents_states:
         if not last_fetch.get(state):
-            last_fetch[state] = FETCH_TIME
+            last_fetch[state] = FIRST_FETCH
 
-    incidents = []
+    for state in incidents_states:
+        if not last_fetched_id.get(state):
+            last_fetched_id[state] = '0'
 
+    incidents = []
     for state in incidents_states:
         request_params = {
             'created_after': last_fetch[state],
-            'state': state
+            'last_fetched_id': last_fetched_id[state],
+            'fetch_delta': fetch_delta,
+            'state': state,
+            'fetch_limit': fetch_limit
         }
-
-        state_parsed_fetch = datetime.strptime(last_fetch[state], TIME_FORMAT)
+        id = last_fetched_id[state]
         incidents_list = get_incidents_batch_by_time_request(request_params)
-        filtered_incidents_list = filter_incidents(incidents_list)
-        for incident in filtered_incidents_list:
-            incident_creation_time = datetime.strptime(incident['created_at'], TIME_FORMAT)
-            if incident_creation_time > state_parsed_fetch:
-                id = incident.get('id')
-                inc = {
-                    'name': 'ProofPoint_TRAP - ID {}'.format(id),
-                    'rawJSON': json.dumps(incident),
-                    'occurred': incident['created_at']
-                }
-                incidents.append(inc)
+        for incident in incidents_list:
+            id = incident.get('id')
+            inc = {
+                'name': 'ProofPoint_TRAP - ID {}'.format(id),
+                'rawJSON': json.dumps(incident),
+                'occurred': incident['created_at']
+            }
+            incidents.append(inc)
 
         if incidents:
             last_fetch_time = incidents[-1]['occurred']
-            last_fetch[state] = last_fetch_time
+            last_fetch[state] = \
+                (datetime.strptime(last_fetch_time, TIME_FORMAT) - timedelta(minutes=1)).isoformat().split('.')[0] + 'Z'
```

Comment:
Why not take the time of the last incident of that state? Couldn't this cause duplicates?

Reply:
No , this will not cause duplicates and also has been tested on a customers instance. The reason is:
I go one minutes back because there is a edge case where the fetch finished (for example due to fetch limit) and the last incident's occured time is x ,However there is another incident whose time is x which wasnt added due to the fetch limit.
As a result the next fetch iteration will start to gather incidents from `created_after` of x and will not bring from the api the incident with the same time which will case missing incidents.

To prevent that I am taking a minute back, however saving the fetched id and bringing only ID's bigger then it, it will prevent duplicate and create the missing incidents . (This works because I sort the incidents from the api by `id` and `created_at` as they are not ordered at all) 
On the next fetch loop we will not bring the other incidents as its time is not bigger then the last fetched one.

There is a unit tests that checks that.